### ISSUE TITLE
Issues with predict gam

### ISSUE THREAD

**Issue by knussearUNR** [rspatial/terra#713]
Using predict gam in this syntax
 tst.r <- terra::predict(myspatraster, mygammodel, type = 'response')
I notice 2 critical issues. 
1) While I have not specified cores (but also when cores = 1) the predict call used multiple cores.
2) The resulting gam comes out with horizontal stripin
![gamissue](https://user-images.githubusercontent.com/107653141/178839374-e3f8241a-a54d-4e03-b98b-6edad64b0834.png)
g

**Comment by rhijmans** [comment]
You do not provide much information (which versoin of "terra", which package you are using to create the GAM; why you think that multiple cores are used.)  But this example appears to work well:

```
library(terra)
logo <- rast(system.file("ex/logo.tif", package="terra"))   
names(logo) <- c("red", "green", "blue")
p <- matrix(c(48, 48, 48, 53, 50, 46, 54, 70, 84, 85, 74, 84, 95, 85, 66, 42, 26, 4, 19, 17, 7, 14, 26, 29, 39, 45, 51, 56, 46, 38, 31, 22, 34, 60, 70, 73, 63, 46, 43, 28), ncol=2)

a <- matrix(c(22, 33, 64, 85, 92, 94, 59, 27, 30, 64, 60, 33, 31, 9, 99, 67, 15, 5, 4, 30, 8, 37, 42, 27, 19, 69, 60, 73, 3, 5, 21, 37, 52, 70, 74, 9, 13, 4, 17, 47), ncol=2)

xy <- rbind(cbind(1, p), cbind(0, a))
e <- extract(logo, xy[,2:3])
v <- data.frame(cbind(pa=xy[,1], e))

library(mgcv)
model <- gam(formula=pa~red+green+blue, data=v)
x <- predict(logo, model)
plot(x)
```

**Comment by knussearUNR** [comment]
My apologies - I was using the current cran version (1.5-34) in R 4.1.2 in  Rstudio 2022.02.3 Build 492. I was also getting messages at the time about "external pointer not valid" at the time. I have since updated to the git version of terra (1.6-1) and things are working now. I have set a temp directory with terraOptions(tempdir = paste0(wd,"/SpatRasterTmps")) - but I do not see any files being written there - so I'm not sure how that is supposed to function to keep the pointers.
